mypy/subtypes.py

```diff
@@ -454,19 +454,22 @@ def visit_instance(self, left: Instance) -> bool:
                     if isinstance(unpacked, Instance):
                         return self._is_subtype(left, unpacked)
             if left.type.has_base(right.partial_fallback.type.fullname):
+                if self.proper_subtype:
+                    return False
```

Comment:
`return False` (here an in few places below) is now correct, but may be error prone if we will add another `if` after this case in the (distant) future. I like how you use `if not self.proper_subtype` + `break`s + `else: return True`. I think it would be better to do the same here (and maybe move the comment below to between `if left.type.has_base()` and `if not self.proper_subtype`).